src/modules/search/components/result/SubscribeToProjectMailinglist.tsx

```diff
@@ -0,0 +1,140 @@
+import React, { useState } from 'react';
+import { Button, IconEye, IconEyeCrossed, Notification } from 'hds-react';
+import { useTranslation } from 'react-i18next';
+import axios from 'axios';
+
+import { Project, DataConfig } from '../../../../types/common';
+import ErrorToast from '../../../../common/errorToast/ErrorToast';
+import useSearchParams from '../../../../hooks/useSearchParams';
+
+import styles from './SubscribeToProjectMailinglist.module.scss';
+
+type Props = {
+  project: Project;
+  config: DataConfig;
+};
+
+enum PostState {
+  Initial,
+  InProgress,
+  Error,
+  Success,
+}
+
+const SubscribeToProjectMailinglist = ({ project, config }: Props) => {
+  const [postState, setPostState] = useState(PostState.Initial);
+  const { t } = useTranslation();
+  const searchParams = useSearchParams();
+  const { housing_company, id } = project;
+
+  const { token, user, static_content } = config;
+
+  const userIsLoggedIn = user.user_id !== 0;
+  const userProfilePageURL = static_content.followed_projects_page_url;
+  const userHasPreviouslySubscribed = user.followed_projects && user.followed_projects.indexOf(id) !== -1;
+
+  const [isSubscribed, setIsSubscribed] = useState(userHasPreviouslySubscribed || false);
+
+  const handleSubscribe = async () => {
+    setPostState(PostState.InProgress);
+    try {
+      const lang = searchParams.get('lang') || 'fi';
+      const result = await axios.post(
+        `/${lang}/project/mailinglist`,
+        {
+          project_id: id,
+        },
+        {
+          headers: {
+            'X-CSRF-TOKEN': token,
+          },
+        }
+      );
+      if (result.data.follow) {
+        setPostState(PostState.Success);
+        setIsSubscribed(result.data.follow);
+      } else {
+        setPostState(PostState.Error);
+      }
```

Comment:
Logic: follow: false is a valid value and unsubscribing always shows error message.

Reply:
Thanks! This is now fixed.